src/App.jsx

```diff
@@ -3,26 +3,91 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './components/api';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    query: '',
+    filter: 'all',
   };
 
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({ todos });
+  }
+
+  async componentDidUpdate() {
+    let todos;
+    const { filter } = this.state;
+
+    if (filter === 'all') {
+      todos = await getTodos();
+    }
+
+    if (filter === 'active') {
+      todos = await getTodos()
+        .then(result => result.filter(todo => todo.completed === false));
+    }
+
+    if (filter === 'completed') {
+      todos = await getTodos()
+        .then(result => result.filter(todo => todo.completed === true));
+    }
+
+    this.pushTodos(todos);
+  }
```

Comment:
Load todos just once and then work with received array.
Your project now is sending infinite requests for todos:
![image](https://user-images.githubusercontent.com/60885495/120671292-ea9b0380-c499-11eb-952e-5046fdd55712.png)